server/routes/unauthenticated/requestRemoval.ts

```diff
@@ -12,34 +15,50 @@ export default class RemovalRequest {
     private readonly systemToken: SystemToken
   ) {}
 
-  view: RequestHandler = async (req, res) => {
+  view: RequestHandler = async (req, res, next) => {
     const statementId = extractStatementId(req)
-    const token = await this.systemToken()
-    const report = await this.reportService.getAnonReportSummary(token, statementId)
+    const { signature } = req.query
 
-    const errors = req.flash('errors')
+    if (!isHashOfString(signature?.toString() || '', statementId.toString(), config.email.urlSigningSecret)) {
+      return next(createHttpError(404, 'Not found'))
```

Comment:
if sig is invalid consider it a 404

Reply:
Could this be simplified? Instead of
`isHashOfString(signature?.toString() || '', statementId.toString(), config.email.urlSigningSecret)`
have something like
`isSignatureValidForStatementId(signature, statementId)`
and push the 'noise' into the function?

Reply:
done this.